velox/external/utf8proc/utf8procImpl.h

```diff
@@ -978,6 +978,15 @@ UTF8PROC_DLLEXPORT int utf8proc_char_length(const char* u_input) {
   return -1;
 }
 
+/// This function is not part of the original utf8proc.
+/// A utf-8 character may be 1 to 4 bytes long. The function will determine if
+/// the input is pointing to the first byte.
+UTF8PROC_DLLEXPORT bool utf8proc_char_first_byte(const char* u_input) {
```

Comment:
Would you explain how this function works?

Reply:
Since the utf-8 code is as follows, when the current byte is 10000000 (128) to 10111111 (191), the current byte is not the first byte of the utf-8 character.

| Number of bytes |  Byte 1  |  Byte 2  |  Byte 3  |  Byte 4  |
| --------------- | :------: | :------: | :------: | :------: |
| 1               | 0xxxxxxx |          |          |          |
| 2               | 110xxxxx | 10xxxxxx |          |          |
| 3               | 1110xxxx | 10xxxxxx | 10xxxxxx |          |
| 4               | 11110xxx | 10xxxxxx | 10xxxxxx | 10xxxxxx |

Reply:
Thank you for clarifying.